test/extensions/transport_sockets/tls/integration/ssl_integration_test.cc

```diff
@@ -87,6 +88,39 @@ void SslIntegrationTestBase::checkStats() {
   counter->reset();
 }
 
+class SslKeyLogTest : public SslIntegrationTest {
+public:
+  SslKeyLogTest(){};
+  void setLocalFilter() {
+    keylog_local_ = true;
+    keylog_remote_ = false;
+  }
+  void setRemoteFilter() {
+    keylog_remote_ = true;
+    keylog_local_ = false;
+  }
+  void setBothLocalAndRemoteFilter() {
+    keylog_local_ = true;
+    keylog_remote_ = true;
+  }
+  void logCheck() {
+#ifdef _MSC_VER
+    std::string log = waitForAccessLog("%temp%\\keylog");
+#else
+    std::string log = waitForAccessLog("/tmp/keylog");
+#endif
+    if (server_tlsv1_3_) {
+      EXPECT_THAT(log, testing::HasSubstr("CLIENT_TRAFFIC_SECRET"));
+      EXPECT_THAT(log, testing::HasSubstr("SERVER_TRAFFIC_SECRET"));
```

Comment:
I htink it would be a stricter validation to check for a regex with all the field names and a pattern for the random parts.

Reply:
I do not think it is necessary to validate more since this log is printed by boringSSL.